web/book/src/reference/declarations/variables.md

```diff
@@ -0,0 +1,73 @@
+# Variables
+
+Variables assign a name to an expression (let's call it `x`). The name can then
+be used in any expression, acting as a substitute for the expression `x`.
+
+Syntactically, variables can take 3 forms.
+
+- `let` declares the name before the expression.
+
+  ```prql no-eval
+  let my_name = x
+  ```
+
+- `into` declares the name after the expression. This form is useful for quick
+  pipeline splitting and conforms with the "flow from top to bottom" rule of
+  pipelines.
+
+  ```prql no-eval
+  x
+  into my_name
+  ```
+
+- implicit name, does not declare a name at all, but uses name `main` as the
+  default. This is practical, as `main` is the variable that is compiled as the
+  main relational query by default.
```

Comment:
Should we clarify it needs to be the final variable in a query? (assuming that's a good way of describing the rule?)

Reply:
Right now, the parser allows it being anywhere in the file, not just the last position. That's why I didn't mention it here. But this should probably be a parser bug, not a language feature